# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dclib/dclib-0.3.23-openssl-1.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Fix compiling when using openssl >= 1.0
Origin: Gentoo Linux

http://bugs.gentoo.org/330291

--- dclib-0.3.23/dclib/core/cssl.cpp
+++ dclib-0.3.23/dclib/core/cssl.cpp
@@ -86,7 +86,11 @@
 /** */
 SSL_CTX * CSSL::InitClientCTX()
 {
+	#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+	const SSL_METHOD *method;
+	#else
 	SSL_METHOD *method;
+	#endif
 	SSL_CTX *ctx = NULL;
 
 	method = SSLv23_client_method();		/* Create new client-method instance */
@@ -109,7 +113,11 @@
 /** */
 SSL_CTX * CSSL::InitServerCTX()
 {
+	#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+	const SSL_METHOD *method;
+	#else
 	SSL_METHOD *method;
+	#endif
 	SSL_CTX *ctx = NULL;
 
 	method = SSLv23_server_method();		/* Create new client-method instance */
@@ -130,7 +138,11 @@
 /** */
 SSL_CTX * CSSL::NewTLSv1ClientCTX()
 {
+	#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+	const SSL_METHOD * method = TLSv1_client_method();
+	#else
 	SSL_METHOD * method = TLSv1_client_method();
+	#endif
 	SSL_CTX * ctx = NULL;
 	
 	if ( method != NULL )
@@ -149,7 +161,11 @@
 /** */
 SSL_CTX * CSSL::NewTLSv1ServerCTX()
 {
+	#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+	const SSL_METHOD * method = TLSv1_server_method();
+	#else
 	SSL_METHOD * method = TLSv1_server_method();
+	#endif
 	SSL_CTX * ctx = NULL;
 	
 	if ( method != NULL )
